airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -301,36 +301,39 @@ def create_custom_object(
     ):
         """
         Creates custom resource definition object in Kubernetes
-
         :param group: api group
         :param version: api version
         :param plural: api plural
         :param body: crd object definition
         :param namespace: kubernetes namespace
         """
         api = client.CustomObjectsApi(self.api_client)
+
         if namespace is None:
             namespace = self.get_namespace()
-        if isinstance(body, str):
-            body_dict = _load_body_to_dict(body)
-        else:
-            body_dict = body
-        try:
-            api.delete_namespaced_custom_object(
-                group=group,
-                version=version,
-                namespace=namespace,
-                plural=plural,
-                name=body_dict["metadata"]["name"],
-            )
-            self.log.warning("Deleted SparkApplication with the same name.")
-        except client.rest.ApiException:
-            self.log.info("SparkApp %s not found.", body_dict['metadata']['name'])
+
+        body_dict: dict = _load_body_to_dict(body) if isinstance(body, str) else body
+
+        # Attribute "name" is not mandatory if "generateName" is used instead
+        if "name" in body_dict["metadata"]:
```

Comment:
can you clarify what you're doing here with this change?

you say
> Attribute "name" is not mandatory if "generateName" is used instead

which suggests, you can still delete the object without `name` as long as `generateName` is present.

but, with your change, if `name` is not present, it appears you won't attempt to delete the object at all!

is it not required to delete the object here? am i missing something?

Reply:
Hi Daniel! Recently there was no functionality of removing already launched objects. So no deleting is ok. Then it was implemented, but the author thought the "name" attribute is mandatory. It is not true -  "generateName" can be used. So we have 2 cases:
1. "name" is used. OK, we can delete the old object if found
2. "name" is not used. It is possible if "generateName" is used. If so, we can not delete the old object - we do not know its name (k8s will generate it randomly using a name mask, but we cant guess it for sure). So we have nothing to do there, the old object will be not touched if exists
Hope this sounds understandable, please ask further to clear the case
Thank you

Reply:
I tihnk it would be great if you add some of that context to commit message. While it links to the issue, it's always good to summarse the context in the commit message so that someone looking at the message will be able to understand the context (rather than having to read the whole issue).